Sources/Compiler/IR/Module.swift

```diff
@@ -1,17 +1,17 @@
 import Utils
 
-/// A module lowered to Val IR.
+/// A collection of IR functions.
 public struct Module {
 
   /// An array of functions.
   public typealias Functions = [Function]
 
+  /// The program defining the functions in `self`.
+  public let program: TypedProgram
```

Comment:
```suggestion
  /// The typechecked but un-lowered source from which `self` was derived.
  public let source: TypedProgram
  ```

Well now, `TypedProgram` isn't really a whole program, is it?

Reply:
Interesting. I would say that it is, and so was `ScopedProgram`. The additional information we gather is only inferred from what we already had.

Reply:
Does every `TypedProgram` have a main?  If not, then it's just a module ersump'n.